### ISSUE TITLE
ecbuild netCDF 

### ISSUE THREAD

**Issue by rhoneyager** [JCSDA/ecbuild#11]
Should be added to #8.

FindNetCDF4.cmake uses nc-config to detect netcdf-fortran, but nc-config's cmake build is used on HomeBrew and on Conda, so the Fortran components are not found. FindNetCDF4.cmake reports success in finding the packages even if the required components are not found.

This, combined with the latest oops and saber merges, breaks jedi builds on macs.

Example from Conda on macOS:
```
nc-config  --all

This netCDF 4.6.2 has been built with the following features: 

  --cc            -> x86_64-apple-darwin13.4.0-clang
  --cflags        -> -I/Users/honeyage/usr/conda/miniconda3/envs/jedi/include
  --libs          -> -L/Users/honeyage/usr/conda/miniconda3/envs/jedi/lib -lnetcdf -lmfhdf -ldf -lhdf5_hl -lhdf5 -lpthread -lz -ldl -lm -lcurl

  --has-c++       -> no
  --cxx           -> 

  --has-c++4      -> no
  --cxx4          -> 

  --has-fortran   -> yes
  --fc            -> nf-config not yet implemented for cmake builds
  --fflags        -> nf-config not yet implemented for cmake builds
  --flibs         -> nf-config not yet implemented for cmake builds
  --has-f90       -> nf-config not yet implemented for cmake builds
  --has-f03       -> nf-config not yet implemented for cmake builds

  --has-dap       -> yes
  --has-dap2      -> yes
  --has-dap4      -> yes
  --has-nc2       -> yes
  --has-nc4       -> yes
  --has-hdf5      -> yes
  --has-hdf4      -> yes
  --has-logging   -> yes
  --has-pnetcdf   -> no
  --has-szlib     -> no
  --has-cdf5      -> yes
  --has-parallel4 -> no
  --has-parallel  -> no

  --prefix        -> /Users/honeyage/usr/conda/miniconda3/envs/jedi
  --includedir    -> /Users/honeyage/usr/conda/miniconda3/envs/jedi/include
  --libdir        -> /Users/honeyage/usr/conda/miniconda3/envs/jedi/lib
  --version       -> netCDF 4.6.2
```

https://github.com/JCSDA/ecbuild/blob/335377a96488c864a27fabee84ea23ae3dd29e28/cmake/contrib/FindNetCDF4.cmake#L157

**Comment by markjolah** [comment]
I see this as mainly a problem with NetCDF's CMake build which has other problems too.  Is it possible to use the autotools NetCDF build proceedure on the mac.  We are currently using that everywhere else.  In any case, my queue is overflowing, and Tom has requested I not work on CMake problems until later this month, so it would have to go in the icebox.

**Comment by rhoneyager** [comment]
Okay. I'll use a workaround for now.

For future reference, here are two vaguely similar scripts that can be adapted to solve the problems here.

[Kitware VTK-provided](https://gitlab.kitware.com/vtk/vtk/blob/master/CMake/FindNetCDF.cmake) - Lacks C++ and Fortran, but adding should be easy.

[FindNetCDF.cmake.txt](https://github.com/JCSDA/ecbuild/files/3687728/FindNetCDF.cmake.txt) - Handles Fortran, but not C++.

**Comment by markjolah** [comment]
Closing as upstream ecbuild has already updated FindNetCDF.cmake which will use a different method to find NetCDF.  Remaining issues should be addressed in #23.
